### DISCUSSION TITLE
Apply an explicit style to an Entry using TextValidationBehavior with InvalidStyle

### DISCUSSION THREAD

**Discussion by Cfun1** [xamarin/XamarinCommunityToolkit#748] [Q&A] [answered]
It seems impossible to apply an explicit style on an `Entry` which is using the `InvalidStyle` of `TextValidationBehavior`, you may say just append/add the explicit style to the `InvalidStyle` definition, but the issue is that the latter is only triggered upon text changed event I believe, thus the style won't be applied all the time.

Any ideas?

## Edit
If I remove the Behaviors SomeEntryStyle is applied, if I put back only InvalidEntryStyle style will be applied when conditions met. the issue here is that SomeEntryStyle  should be applied regardless.

I am not sure if I am missing something or something needs to change in xct.
```
      <Entry Text="text"
                   Placeholder="palceholder"
                   Style="{StaticResource SomeEntryStyle}">
                <Entry.Behaviors>
                    <xct:EmailValidationBehavior DecorationFlags="Trim"
                                                 InvalidStyle="{StaticResource InvalidEntryStyle}"/>
                </Entry.Behaviors>
            </Entry>
```
```
    <ContentPage.Resources>
        <Style x:Key="SomeEntryStyle"
               TargetType="Entry">
            <Setter Property="TextColor" Value="White"/>
            <Setter Property="PlaceholderColor" Value="Blue"/>
            <Setter Property="BackgroundColor" Value="Purple"/>
            <Setter Property="Visual" Value="Material"/>
        </Style>

        <Style x:Key="InvalidEntryStyle"
               TargetType="Entry">
            <Setter Property="TextColor" Value="Red"/>
        </Style>
    </ContentPage.Resources>
```

**Comment by AndreiMisiukevich** [comment]
Hello,
if I got you correctly: Any validation behavior has the property "Flags". By setting this property you can manage when/how validation behavior should update the state.

https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/Validators/ValidationFlags.shared.cs#L6-L13

Choose your strategy of validation.

**Reply by Cfun1** [reply]
No it is not what I meant, my bad I should have included a concrete example, please check I have update the initial question.

**Comment by Cfun1** [comment, marked as answer]
Moved to an issue https://github.com/xamarin/XamarinCommunityToolkit/issues/807
